### ISSUE TITLE
Consider option to push page content

### ISSUE THREAD

**Issue by tonygentilcore** [balloob/react-sidebar#74]
Realizing this would be a departure from the MD spec, would this project consider a disabled by default option to push page content as the sidebar opens rather than overlap it? That's hard to word, see the "push" option of http://negomi.github.io/react-burger-menu/ for example.

I have a hacky patch that does this, but wanted to check whether it'd be accepted before spending time cleaning it up.

**Comment by balloob** [comment]
This is what the `docked` property does. See the demo: http://balloob.github.io/react-sidebar/example/

**Comment by tonygentilcore** [comment]
Perfect! Apologies for the noise.

**Comment by tonygentilcore** [comment]
So, actually there's one wrinkle. While toggling `docked` on that page gives the _animation_ I'm looking for, I'd like the same _touch behavior_ as toggling `open`.

**Comment by balloob** [comment]
That seems like a weird user experience. Would you expect the content to resize while you are dragging the sidebar left-right ? 

I also think that it would be very expensive because the content will have to resize without the browser able to optimize it, many times per second. In the case of `docked` we use a CSS transition + `will-change: left` (oh, actually I see now that we don't do the latter).

**Comment by tonygentilcore** [comment]
No, I definitely wouldn't want resizing (you're right that'd be a bad user experience as well as slow). I'd like clipping (same behavior as react-burger's "push" animation). But I *think* I can clip by overriding default styles from outside of this component.

**Comment by martin-braun** [comment]
> That seems like a weird user experience. Would you expect the content to resize while you are dragging the sidebar left-right ?

Actually, take a look at this page on smaller window width: https://ant.design/components/grid/ Take inspection to the side navigation drawer they have there, it pushes the content of the page without resizing it. I think this is what @tonygentilcore wants and I'd love to see an option for that, too.

**Comment by ivan-aksamentov** [comment]
Another example of "push" effect is illustrated in [`react-burger-menu`](https://negomi.github.io/react-burger-menu/) demo (option "push"). 

`react-burger-menu` however does not support the "responsive sticky" feature that `react-sidebar` has. And I desperately want both.
